### ISSUE TITLE
linter: false positive in `no-accumulating-spread` when nesting functions inside reduce

### ISSUE THREAD

**Issue by sbalay** [oxc-project/oxc#14894]
### What version of Oxlint are you using?

1.23.0

### What command did you run?

oxlint --type-aware --report-unused-disable-directives-severity=error

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["typescript", "unicorn", "oxc", "vitest", "import", "promise"],
  "ignorePatterns": [
    "dist",
    "node_modules",
    "build",
    "coverage",
  ],
  "categories": {
    "correctness": "error",
    "perf": "error",
    "suspicious": "error"
  }
}

```

### What happened?

Consider the following code (weird and not realistic but hope you get the point)

```ts
function doSomething(
  list: any[]
) {
  return list.reduce((acc, each) => {
    return each.subList.flatMap((subEach) => {
        return [...acc, subEach.subList]
    })
  }, [])
}
```

oxlint will report `no-accumulating-spread` as an error and suggest:

> Do not spread accumulators in Array.prototype.reduce()
help: It looks like you're spreading an `Array`. Consider using the `Array.push` or `Array.concat` methods to mutate the accumulator instead.
Using spreads within accumulators leads to `O(n^2)` time complexity.oxcoxc(no-accumulating-spread)

However, actually mutating the list as suggested would lead to a change in behavior and most likely result in bugs:

```ts
function doSomething(
  list: any[]
) {
  return list.reduce((acc, each) => {
    return each.subList.flatMap((subEach) => {
        acc.push(subEach) // mutates the list as suggest, will probably result in bugs.
        return acc
    })
  }, [])
}
```

**Comment by camc314** [comment]
This took me a while to wrap my head around. but thank you for finding it.
